cmd/adminkubeconfig/get.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/redhatopenshift"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+var (
+	gitCommit    = "unknown"
+	requiredEnvs = []string{
```

Comment:
by taking the cluster resourceId on the command line and by using NewAuthorizerFromEnvironment, you can remove all of the requiredEnvs boilerplate.